drivers/platform/aducm3029/gpio.c

```diff
@@ -0,0 +1,236 @@
+ */
+int32_t gpio_direction_input(struct gpio_desc *desc)
+{
+	if (!desc || !nb_gpio)
+		return FAILURE;
+	/* Enable input driver */
+	if (ADI_GPIO_SUCCESS != adi_gpio_InputEnable(PORT(desc->number),
+			PIN(desc->number), true))
+		return FAILURE;
+	return SUCCESS;
+}
+
+/**
+ * @brief Enable the output direction of the specified GPIO and set the GPIO to
+ * the specified value
+ * @param desc - The GPIO descriptor.
+ * @param value - The value. \ref GPIO_HIGH or \ref GPIO_LOW
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t gpio_direction_output(struct gpio_desc *desc, uint8_t value)
+{
+	ADI_GPIO_RESULT ret;
+
+	if (!desc || !nb_gpio)
+		return FAILURE;
+
+	/* Enable output driver */
+	ret = adi_gpio_OutputEnable(PORT(desc->number), PIN(desc->number),
+				    true);
+	if(ret != ADI_GPIO_SUCCESS)
+		return FAILURE;
+
+	/* Initialize pin with a value */
+	if(value == 1)
+		ret = adi_gpio_SetHigh(PORT(desc->number), PIN(desc->number));
+	else
+		ret = adi_gpio_SetLow(PORT(desc->number), PIN(desc->number));
+	if(ret != ADI_GPIO_SUCCESS)
+		return FAILURE;
+	return SUCCESS;
+}
+
+/**
+ * @brief Get the direction of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param direction - Variable where to store the direction. Will be set to \ref
+ * GPIO_OUT or \ref GPIO_IN
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t gpio_get_direction(struct gpio_desc *desc, uint8_t *direction)
+{
+	uint16_t pins;
+
+	if (!desc || !nb_gpio)
+		return FAILURE;
+
+	if (ADI_GPIO_SUCCESS != adi_gpio_GetOutputEnable(PORT(desc->number),
+			&pins))
+		return FAILURE;
+	if (pins & PIN(desc->number))
+		*direction = GPIO_OUT;
+	else
+		*direction = GPIO_IN;
+	return SUCCESS;
+}
+
+/**
+ * @brief Set the value of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param value - The value: GPIO_HIGH or GPIO_LOW
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t gpio_set_value(struct gpio_desc *desc, uint8_t value)
+{
+	ADI_GPIO_RESULT ret;
+
+	if (!desc || !nb_gpio)
+		return FAILURE;
+
+	if(value == GPIO_HIGH)
+		ret = adi_gpio_SetHigh(PORT(desc->number), PIN(desc->number));
+	else
+		ret = adi_gpio_SetLow(PORT(desc->number), PIN(desc->number));
+	if (ret != ADI_GPIO_SUCCESS)
+		return FAILURE;
+	return SUCCESS;
+}
+
+/**
+ * @brief Get the value of the specified GPIO.
+ * @param desc - The GPIO descriptor.
+ * @param value - Variable where to store the direction. Will be set to \ref
+ * GPIO_HIGH or \ref GPIO_LOW
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t gpio_get_value(struct gpio_desc *desc, uint8_t *value)
+{
+	uint16_t pins;
+
+	if (!desc || !nb_gpio)
+		return FAILURE;
+
+	if (ADI_GPIO_SUCCESS != adi_gpio_GetData(PORT(desc->number),
+			PIN(desc->number), &pins))
+		return FAILURE;
+	if (pins & PIN(desc->number))
+		*value = GPIO_HIGH;
+	else
+		*value = GPIO_LOW;
```

Comment:
As it stands `adi_gpio_GetData` can only be accessed with one pin. This means that it should be OK to write:
`*value = pins;`
or to be safer and still compress the code:
`*value = !!pins;`

Reply:
true